components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/PreferenceRetrievalClient.java

```diff
@@ -264,17 +314,17 @@ public boolean checkPreference(String tenant, String connectorName, String prope
                         JSONObject config = responseProperties.getJSONObject(itemIndex);
                         if (StringUtils.equalsIgnoreCase(
                                 responseProperties.getJSONObject(itemIndex).getString(PROPERTY_NAME), propertyName)) {
-                            return Boolean.valueOf(config.getString(PROPERTY_VALUE));
+                            return config.getString(PROPERTY_VALUE);
                         }
                     }
                 }
-                return defaultValue;
```

Comment:
where was this previously obtained from?

Reply:
https://github.com/wso2/carbon-identity-framework/blob/12a151508ff2b70a174abc3973f63da851b39985/components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/PreferenceRetrievalClient.java#L257